### PR TITLE
Quill Locker/ add team lockers

### PR DESCRIPTION
**Pull request by eadams17**
## WHAT
this is the first version of Quill Locker! There is a locker for each team and an "All lockers" section that has every locker listed

## WHY
we want to improve the UI of internal tooling for staff

## HOW
created a new app called "locker" in the `Staff` bundle with several new components: `LockerIndex`, `TeamLocker` and `Locker` as well a bunch of new consts used in a `lockerItems` file which houses all of the data for each locker. 

### Screenshots
(If applicable. Also, please censor any sensitive data)
Screen Shot 2022-02-23 at 7 02 34 PM
Screen Shot 2022-02-23 at 7 02 42 PM

### Notion Card Links
(Please provide links to any relevant Notion card(s) relevant to this PR.)
https://www.notion.so/quill/Organize-internal-tooling-into-team-categories-5752e0e2c9b74a1e8616f3dbe12b22e2

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? |  yes
Have you deployed to Staging? | yes
Self-Review: Have you done an initial self-review of the code below on Github? | yes
Spec Review: Have you reviewed the spec and ensured this meets requirements and/or matches design mockups? | yes

### CODE REVIEW

File: services/QuillLMS/client/app/bundles/Staff/components/locker/lockerIndex.tsx [hunk 1]
```diff
@@ -1,7 +1,26 @@
...
+  const allLockers = [...TEAMS, 'index'];
+  function renderLockers() {
+    return allLockers.map((lockerKey: string) => (
+      <Locker key={lockerKey} lockerContents={lockerItems[lockerKey]} />
+    ));
+  }
+
+  return(
+    <div className="locker-index-container locker-content">
+      <h3 className="subheader">Team Lockers</h3>
```
Comment (emilia-friedberg) [inline services/QuillLMS/client/app/bundles/Staff/components/locker/lockerIndex.tsx]: Can we sentence case this for consistency?

File: services/QuillLMS/client/app/bundles/Staff/components/locker/teamLocker.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    if(label === 'all lockers') { return <h3 className="subheader">Index</h3>}
+    return <h3 className="subheader">{`${titleCase(label)} lockers`}</h3>
+  }
+
+  return(
+    <div className="team-locker-container locker-content">
+      <Link className="back-nav-link" tabIndex={-1} to="/locker">
+        <button className="button-container interactive-wrapper focus-on-light">
+          <img alt="left pointing arrow" src={arrow} />
+          <p>All Lockers</p>
```
Comment (emilia-friedberg) [inline services/QuillLMS/client/app/bundles/Staff/components/locker/teamLocker.tsx]: Sentence case?

File: services/QuillLMS/client/app/bundles/Staff/components/locker/teamLocker.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+  function renderHeader() {
+    if(label === 'all lockers') { return <h3 className="subheader">Index</h3>}
+    return <h3 className="subheader">{`${titleCase(label)} lockers`}</h3>
+  }
+
+  return(
+    <div className="team-locker-container locker-content">
+      <Link className="back-nav-link" tabIndex={-1} to="/locker">
+        <button className="button-container interactive-wrapper focus-on-light">
```
Comment (emilia-friedberg) [inline services/QuillLMS/client/app/bundles/Staff/components/locker/teamLocker.tsx]: Can't we just apply this class to the `<Link>`? This is an internal tool so it isn't a huge deal, but it's an accessibility issue to have a negative tabindex on an interactive element and no interaction on a `button` element, and we might one day have someone on staff who uses a screenreader.

**Review by emilia-friedberg — APPROVED** [review]
This looks SO GOOD. Obsessed with your emoji choices, design looks clean and soothing, so excited to use! Since it is so professional looking already I just made some notes about standardizing case and one about a potential accessibility issue, but I don't need to look at it again. Yay Eric!!!

File: services/QuillLMS/client/app/bundles/Shared/libs/titleCase.ts [hunk 4]
```diff
@@ -0,0 +1,3 @@
+export function titleCase(string: string){
+  return string[0].toUpperCase() + string.slice(1).toLowerCase();
```
Comment (cissyxyu) [inline services/QuillLMS/client/app/bundles/Shared/libs/titleCase.ts]: Maybe add an edge case here for when string is null or empty string? I know it's probably rare but if we end up using this method later down the line for other functions we might want it to be fail-safe.

**Review by cissyxyu — APPROVED** [review]
Looks great! Excited to start using it :)

File: services/QuillLMS/client/app/bundles/Staff/components/locker/teamLocker.tsx (hunk 3 above)
Comment (eadams17) [reply services/QuillLMS/client/app/bundles/Staff/components/locker/teamLocker.tsx]: good call-- I didn't realize the negative tab index would mess with that. I'll remove the `Link` element and just add an `onClick` handler on the button so we can use the accessible focus indicator on the button

**Pull request merged**